packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts

```diff
@@ -0,0 +1,511 @@
+            // this container does not have access to the blob's handle.
+            const localResponse = await container2.request({ url: localHandle.absolutePath });
+            assert.strictEqual(localResponse?.status, 404, `Expecting a 404 response for local handle`);
+            assert(localResponse.value.startsWith("Blob removed by gc:"), `Unexpected value for local handle 1`);
+
+            // Retrieving the blob via the storage handle should fail as well.
+            const storageResponse = await container2.request({ url: storageHandle.absolutePath });
+            assert.strictEqual(storageResponse?.status, 404, `Expecting a 404 response for storage handle`);
+            assert(storageResponse.value.startsWith("Blob removed by gc:"), `Unexpected value for local handle 2`);
+        });
+
+        itExpects("tombstones blobs uploaded and de-duped in detached container",
+        [
+            {
+                eventName: "fluid:telemetry:BlobManager:GC_Tombstone_Blob_Requested",
+            },
+            {
+                eventName: "fluid:telemetry:BlobManager:GC_Tombstone_Blob_Requested",
+            },
+            {
+                eventName: "fluid:telemetry:BlobManager:GC_Tombstone_Blob_Requested",
+            }
+        ],
+        async () => {
+            // Upload couple of attachment blobs with the same content. When these blobs are uploaded to the server,
+            // they will be de-duped and redirect to the same storageId.
+            const blobContents = "Blob contents";
+            const localHandle1 = await mainDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+            const localHandle2 = await mainDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Attach the container after the blob is uploaded.
+            await mainContainer.attach(provider.driver.createCreateNewRequest(provider.documentId));
+
+            // Send an op to transition the container to write mode.
+            mainDataStore._root.set("transition to write", "true");
+            await waitForContainerConnection(mainContainer);
+
+            const summarizer = await createSummarizer(
+                provider,
+                mainContainer,
+                undefined /* summaryVersion */,
+                gcOptions,
+                mockConfigProvider(settings),
+            );
+
+            // Add the blob's local handles to reference them.
+            mainDataStore._root.set("localBlob1", localHandle1);
+            mainDataStore._root.set("localBlob2", localHandle2);
+
+            // Upload the same blob. This will get de-duped and we will get back a handle with the storageId instead of
+            // the localId that we got when uploading in detached container.
+            const storageHandle = await mainDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
```

Comment:
Why doesn't the upload happen automatically? And why is it that we're only uploading localBlob1 and not localBlob2

Reply:
I don't follow what you are saying. What do you mean upload automatically? Both localHandle1 and localHandle2 are results of uploading blobs.